c10/core/TensorImpl.cpp

```diff
@@ -526,8 +526,11 @@ c10::intrusive_ptr<TensorImpl> TensorImpl::shallow_copy_and_detach_core(
       /*dest_impl=*/impl.get(),
       /*version_counter=*/std::forward<VariableVersion>(version_counter),
       /*allow_tensor_metadata_change=*/allow_tensor_metadata_change);
-  impl->refresh_numel();
-  impl->refresh_contiguous();
+  if (impl->sizes_strides_policy_ <
+      static_cast<uint8_t>(SizesStridesPolicy::CustomSizes)) {
+    impl->refresh_numel();
+    impl->refresh_contiguous();
+  }
```

Comment:
No, this should be conditioned on `has_symbolic_sizes_strides`. Otherwise this is BC breaking for subclasses that set CustomSizes but explicitly call `default_numel` in their implementation; the `numel_` field is now garbage.

Reply:
done

Reply:
Done after some discussion (and thanks to Ed for figuring out the additional fix!)